### PR TITLE
[MAINTENANCE] Ensure that `id_` fields in Marshmallow schema serialize as `id`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Ensure that all cases of `id_` in our Marshmallow schema properly serialize as `id`

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -2239,8 +2240,27 @@ class CheckpointValidationConfigSchema(Schema):
     class Meta:
         unknown = INCLUDE
 
-    name = fields.String(required=False, allow_none=False)
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:2242]: Should have been removed in a prior PR but forgot to.

File: great_expectations/data_context/types/base.py [hunk 2]
```diff
@@ -2311,7 +2331,9 @@ class Meta:
     evaluation_parameters = fields.Dict(required=False, allow_none=True)
     runtime_configuration = fields.Dict(required=False, allow_none=True)
     validations = fields.List(
-        cls_or_instance=fields.Dict(), required=False, allow_none=True
+        cls_or_instance=fields.Nested(CheckpointValidationConfigSchema),
+        required=False,
+        allow_none=True,
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:2337]: Never converted this over to the typed obj - thanks @Kilo59!

File: great_expectations/data_context/types/base.py [hunk 3]
```diff
@@ -2239,8 +2240,27 @@ class CheckpointValidationConfigSchema(Schema):
     class Meta:
         unknown = INCLUDE
 
-    name = fields.String(required=False, allow_none=False)
-    id_ = fields.String(required=False, allow_none=False)
+    id_ = fields.String(required=False, allow_none=False, data_key="id")
+
+    def dump(self, obj: dict, *, many: Optional[bool] = None) -> dict:
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:2245]: Do we have an explicit list of what can go in validations nested within checkpoints? If so, we can get rid of this. As this has historically been a dictionary, I wanted to guard against edge cases and make this as forgiving of unknown values as possible.

**Review by Kilo59 — APPROVED** [review]

File: great_expectations/data_context/types/base.py (hunk 3 above)
Comment (anthonyburdi) [reply great_expectations/data_context/types/base.py:2245]: I did not find a list in a quick search, but what about using all of the keys in validations from https://docs.greatexpectations.io/docs/guides/validation/checkpoints/how_to_configure_a_new_checkpoint_using_test_yaml_config and https://docs.greatexpectations.io/docs/guides/validation/checkpoints/how_to_add_validations_data_or_suites_to_a_checkpoint as a starting point?

**Review by anthonyburdi — APPROVED** [review]
LGTM

File: great_expectations/data_context/types/base.py (hunk 3 above)
Comment (anthonyburdi) [reply great_expectations/data_context/types/base.py:2245]: Also take a look at get_substituted_validation_dict().
Maybe best to create the list but also leave it open until we can make sure we have them all.

**Review by anthonyburdi — APPROVED** [review]

**Pull request merged**